realscale.py

```diff
@@ -101,10 +180,20 @@ def effect(self):
         
         # calculate scaling center
         center = self.calc_scale_center(p1_x, p1_y, p2_x, p2_y)
+        drawnscale = int(self.options.drawnscale)
 
         #calculate scaling factor
         target_length = self.getUnittouu(str(self.options.length) + self.options.unit)
-        factor = target_length / p_length
+        #inkex.debug("%s, %s" % (target_length, p_length))
+        factor = (target_length / p_length) / drawnscale
+        
+        # Add scale rule
+        if self.options.showscale == "true":
+            dist = int(self.options.unitlength)
+            # !! this isn't working right if drawnscale != 1 (1:1)
```

Comment:
This part has abug but my mind has gone :(
Works perfectly I think if scale is 1:1
but ruler seems wrong size if scale larger... confused...